notebook/nbextensions.py

```diff
@@ -299,118 +414,356 @@ def _config_file_name_default(self):
+class UninstallNBExtensionApp(BaseNBExtensionApp):
+    """Entry point for uninstalling notebook extensions"""
     version = __version__
-    description = "Configure an nbextension to be automatically loaded"
+    description = """Uninstall Jupyter notebook extensions
+    
+    Usage
+    
+        jupyter nbextension uninstall path/url path/url/entrypoint
+        jupyter nbextension uninstall --py pythonPackageName
+    
+    This uninstalls an nbextension.
+    """
+    
+    examples = """
+    jupyter nbextension uninstall dest/dir dest/dir/extensionjs
+    jupyter nbextension uninstall --py extensionPyPackage
+    """
+    aliases = {'section': 'ToggleNBExtensionApp.section'}
+    flags = _uninstall_flags
+    
+    user = Bool(False, config=True, help="Whether to do a user install")
+    sys_prefix = Bool(False, config=True, help="Use the sys.prefix as the prefix")
 
-    section = Unicode('notebook', config=True,
-          help=("Which config section to add the extension to. "
-                "'common' will affect all pages.")
+    prefix = Unicode('', config=True, help="Installation prefix")
+    nbextensions_dir = Unicode('', config=True, help="Full path to nbextensions dir (probably use prefix or user)")
+    destination = Unicode('', config=True, help="Destination for the copy or symlink")
+    python = Bool(False, config=True, help="Uninstall nbextension in a Python package")
+    verbose = Enum((0,1,2), default_value=1, config=True,
+        help="Verbosity level"
     )
 
-    aliases = {'section': 'EnableNBExtensionApp.section',
-              }
-
     def _config_file_name_default(self):
         return 'jupyter_notebook_config'
-
-    def enable_nbextension(self, name):
-        # Local import to avoid circular import issue on Py 2
-        from .services.config import ConfigManager
-        cm = ConfigManager(parent=self, config=self.config)
-        cm.update(self.section, {"load_extensions": {name: True}})
-
+    
+    def uninstall_extensions(self):
+        kwargs = {
+            'verbose': self.verbose,
+            'user': self.user,
+            'sys_prefix': self.sys_prefix,
+            'prefix': self.prefix,
+            'nbextensions_dir': self.nbextensions_dir,
+        }
+        
+        if self.python:
+            if len(self.extra_args)>1:
+                raise ValueError("only one nbextension allowed at a time.  Call multiple times to uninstall multiple extensions.")
+            if len(self.extra_args)<1:
+                raise ValueError("not enough arguments")
+            uninstall_nbextension_python(self.extra_args[0], **kwargs)
+        else:
+            if len(self.extra_args)>2:
+                raise ValueError("only one nbextension allowed at a time.  Call multiple times to uninstall multiple extensions.")
+            if len(self.extra_args)<2:
+                raise ValueError("not enough arguments")
+            uninstall_nbextension(self.extra_args[0], self.extra_args[1], **kwargs)
```

Comment:
move the `if len raise`/`if len raise` before the `if self.python`, they are the same.